src/Sulu/Bundle/MediaBundle/Resources/js/stores/CollectionStore/CollectionStore.js

```diff
@@ -1,78 +1,101 @@
 // @flow
-import {action, autorun, computed, observable} from 'mobx';
+import {computed} from 'mobx';
 import type {IObservableValue} from 'mobx'; // eslint-disable-line import/named
-import {ResourceRequester} from 'sulu-admin-bundle/services';
-import type {BreadcrumbItem, BreadcrumbItems, Collection} from './types';
+import {ResourceStore} from 'sulu-admin-bundle/stores';
+import type {BreadcrumbItem, BreadcrumbItems} from './types';
 
 const COLLECTIONS_RESOURCE_KEY = 'collections';
 
 export default class CollectionStore {
-    @observable loading: boolean = false;
-    @observable collection: Collection = {
-        id: null,
-        parentId: null,
-        breadcrumb: null,
-    };
-    disposer: () => void;
+    collectionId: ?string | number;
+    locale: IObservableValue<string>;
+    resourceStore: ResourceStore;
 
     constructor(collectionId: ?string | number, locale: IObservableValue<string>) {
-        this.disposer = autorun(() => {
-            this.load(collectionId, locale.get());
-        });
+        this.collectionId = collectionId;
+        this.locale = locale;
+        this.resourceStore = new ResourceStore(
+            COLLECTIONS_RESOURCE_KEY,
+            collectionId,
+            {
+                locale,
+            },
+            {
+                depth: 1,
+                breadcrumb: true,
+            }
+        );
     }
 
     destroy() {
-        this.disposer();
+        this.resourceStore.destroy();
     }
 
-    @computed get id(): ?number {
-        return this.collection.id;
+    create(options: Object = {}) {
+        return this.resourceStore.save(options);
     }
 
-    @computed get parentId(): ?number {
-        return this.collection.parentId;
+    update(options: Object = {}) {
```

Comment:
Why don't we simply have a `save` method? That was also something with the API right? I think we should fix it in the right place, instead of going away from the optimal API here.